src/utils/sentry/httpclient.ts

```diff
@@ -0,0 +1,486 @@
+      },
+    })
+
+    captureEvent(event)
+  }
+}
+
+/**
+ * Interceptor function for XHR requests
+ *
+ * @param xhr The XHR request
+ * @param method The HTTP method
+ * @param headers The HTTP headers
+ */
+function _xhrResponseHandler(
+  options: HttpClientOptions,
+  xhr: SentryWrappedXMLHttpRequest & XMLHttpRequest,
+  method: string,
+  headers: Record<string, string>,
+): void {
+  const shouldCapture = _shouldCaptureResponse(options, xhr.status, xhr.responseURL)
+
+  if (shouldCapture) {
+    let requestHeaders, responseCookies, responseHeaders
+
+    if (_shouldSendDefaultPii()) {
+      try {
+        const cookieString =
+          xhr.getResponseHeader('Set-Cookie') || xhr.getResponseHeader('set-cookie') || undefined
+
+        if (cookieString) {
+          responseCookies = _parseCookieString(cookieString)
+        }
+      } catch (e) {
+        logger.log('Could not extract cookies from response headers')
+      }
+
+      try {
+        responseHeaders = _getXHRResponseHeaders(xhr)
+      } catch (e) {
+        logger.log('Could not extract headers from response')
+      }
+
+      requestHeaders = headers
+    }
+
+    const event = _createEvent({
+      request: {
+        url: xhr.responseURL,
+        method,
+        requestHeaders,
+        query_string: xhr.__sentry_xhr_v3__?.url.includes('?')
+          ? xhr.__sentry_xhr_v3__.url.split('?')[1]
+          : undefined,
+        data: xhr.__sentry_xhr_v3__?.body,
+      },
+      response: {
+        // Can't access request cookies from XHR
+        responseHeaders,
+        responseCookies,
+        status: xhr.status,
+      },
+    })
+
+    captureEvent(event)
+  }
+}
+
+/**
+ * Extracts response size from `Content-Length` header when possible
+ *
+ * @param headers
+ * @returns The response size in bytes or undefined
+ */
+function _getResponseSizeFromHeaders(headers?: Record<string, string>): number | undefined {
+  if (headers) {
+    const contentLength = headers['Content-Length'] || headers['content-length']
+
+    if (contentLength) {
+      return parseInt(contentLength, 10)
+    }
+  }
+
+  return undefined
+}
+
+/**
+ * Creates an object containing cookies from the given cookie string
+ *
+ * @param cookieString The cookie string to parse
+ * @returns The parsed cookies
+ */
+function _parseCookieString(cookieString: string): Record<string, string> {
+  return cookieString.split('; ').reduce((acc: Record<string, string>, cookie: string) => {
+    const [key, value] = cookie.split('=')
+    acc[key] = value
+    return acc
+  }, {})
+}
+
+/**
+ * Extracts the headers as an object from the given Fetch API request or response object
+ *
+ * @param headers The headers to extract
+ * @returns The extracted headers as an object
+ */
+function _extractFetchHeaders(headers: Headers): Record<string, string> {
+  const result: Record<string, string> = {}
+
+  headers.forEach((value, key) => {
+    result[key] = value
+  })
+
+  return result
```

Comment:
Be cooler if you did

```suggestion
  return Object.fromEntries(headers)
```

Reply:
Meh, that's part of the original file and would rather keep it as-is for diffing purposes!